### PR TITLE
adding mona dataset

### LINKED ISSUE #137: New Task: Adding Data from MassBank of North America (MoNA)
This dataset can be cross-referenced later on when we add HMDB (#136)

### PR DESCRIPTION
**Pull request by adamoyoung**
fixes #137

I have uploaded the data to [my personal huggingface account](https://huggingface.co/datasets/adamoyoung/mona), but am happy to move it over to OpenBioML.

I have included the [data preprocessing notebook](data/mona/example_processing_and_templates.ipynb) since there is some non-trivial data processing.

I have some ideas on how to set up queries for this dataset, and have provided a few examples in the meta file.

### CODE REVIEW

**Comment by kjappelbaum** [comment]
Thanks for your contribution and sorry for letting you wait!

@MicPie can you help with the huggingface?

**Comment by MicPie** [comment]
@adamoyoung Can you also commit the `meta.yaml` file?
Before we move the HF dataset to OpenBioML we can add the infos from the `meta.yaml` there.
Let me know if you need help there. 😃

**Comment by adamoyoung** [comment]
@MicPie great, just to confirm you mean commit the meta.yaml here (and not in the huggingface dataset repo?)

**Comment by MicPie** [comment]
@adamoyoung
> great, just to confirm you mean commit the meta.yaml here (and not in the huggingface dataset repo?)
Yes, here in the repo. 👍

File: data/mona/transform.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
+import urllib
+import urllib.request
+
+import pandas as pd
+import yaml
+
+DATASET_URL = (
+    "https://huggingface.co/datasets/adamoyoung/mona/resolve/main/data/mona_df.json"
+)
+META_YAML_PATH = "./data/mona/meta.yaml"
```
Comment (MicPie) [inline data/mona/transform.py:10]:
```suggestion
META_YAML_PATH = __file__.replace("transform.py","meta.yaml")
```
This should make it work no matter from which dir you run it.

**Review by MicPie — CHANGES_REQUESTED** [review]
Hi @adamoyoung 
Thank you for adding the `meta.yaml` to the PR.
Just one thing is missing: The `transform.py` is not saving the raw data (see for example here: https://github.com/OpenBioML/chemnlp/blob/main/data/ames_mutagenicity/transform.py#L53:L54).
Once this works end-to-end we can also move the HF dataset to the OpenBioML org.
Please let me know if you have time to fix that, otherwise we can jump in.

**Comment by adamoyoung** [comment]
@MicPie does this address your comments? I save the dataframe as a csv at `data/mona/data_clean.csv`. https://github.com/OpenBioML/chemnlp/blob/7e3c48bce214b11c44bda2026ab283766fc28124/data/mona/transform.py#L11

**Comment by adamoyoung** [comment]
@MicPie is it ready to merge?

**Comment by MicPie** [comment]
Hi @adamoyoung 
Again sorry for the late reply, this now looks very good!
The only thing that is not working is that `http://semanticscience.org/resource/CHEMINF_000417` does not resolve.
I'm checking that.
If you have time feel free to copy some information from the `meta.yaml` to the HF dataset card so we have some information there (especially the license). If you are busy, just tell me then I can do that after you moved the dataset.
So just let me know and thank you for your help! :-D

**Pull request merged**